sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -1009,35 +1005,4 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
       case _ =>
     }
   }
-
-  // Make sure that the `SHOW PARTITIONS` command is allowed for the table
-  private def checkShowPartitions(showPartitions: ShowPartitions): Unit = showPartitions match {
-    case ShowPartitions(rt: ResolvedTable, _, _)
-        if !rt.table.isInstanceOf[SupportsPartitionManagement] =>
-      failAnalysis("SHOW PARTITIONS cannot run for a table which does not support partitioning")
-    case ShowPartitions(ResolvedTable(_, _, partTable: SupportsPartitionManagement, _), _, _)
-        if partTable.partitionSchema().isEmpty =>
-      failAnalysis(
-        s"SHOW PARTITIONS is not allowed on a table that is not partitioned: ${partTable.name()}")
-    case _ =>
-  }
-
-  private def checkTruncateTable(truncateTable: TruncateTable): Unit = truncateTable match {
-    case TruncateTable(rt: ResolvedTable, None) if !rt.table.isInstanceOf[TruncatableTable] =>
```

Comment:
This check and below guarded from `scala.MatchError` in `run()`. How does the contract work now?

Reply:
I have updated the execution part, it will throw proper error instead of MatchError.